src/QsCompiler/DataStructures/ReservedKeywords.fs

```diff
@@ -239,6 +238,28 @@ module AssemblyConstants =
     let AlfredProcessor = "AlfredProcessor"
     let BrunoProcessor = "BrunoProcessor"
     let ClementineProcessor = "ClementineProcessor"
+    let DefaultSimulator = "DefaultSimulator"
+    let QuantumSimulator = "QuantumSimulator"
+    let ToffoliSimulator = "ToffoliSimulator"
+    let ResourcesEstimator = "ResourcesEstimator"
+
+    // Note: The names of the capabilities here need to match the ones defined by the Sdk.
+    type RuntimeCapabilities = 
+    | Unknown = 0 
+    | QPRGen0 = 1
+    | QPRGen1 = 2
+
+
+/// contains reserved names for command line arguments
+module CommandLineArguments = 
+    let SimulatorOption = "simulator"
+    let ReservedArguments = ImmutableArray.Create SimulatorOption
+    let ReservedArgumentAbbreviations = ImmutableArray.Create SimulatorOption.[0] 
+
+    let BuiltInSimulators =
+        // Note: If this is adapted, then the error message for InvalidExecutionTargetForTest needs to be adapted as well.
+        [AssemblyConstants.QuantumSimulator; AssemblyConstants.ToffoliSimulator; AssemblyConstants.ResourcesEstimator]
+        |> ImmutableHashSet.CreateRange
```

Comment:
It might be better to have the error message convert `BuiltInSimulators` to a string directly, so the list is not stored in two places.

Reply:
Hmm, thinking about it, I removed the comment, because they might not exactly match and if we were to enable an additional simulator, we should have the proper unit tests to test that functionality in any case.